### PR TITLE
feat/fix-platform-key-annotation

### PR DESCRIPTION
**Pull request by ahoeing**

### CODE REVIEW

File: sda-commons-server-key-mgmt/src/main/java/org/sdase/commons/keymgmt/validator/PlatformKey.java [hunk 1]
```diff
@@ -22,7 +23,7 @@
  *
  * <p>The validator accepts <code>null</code> values.
  */
-@Target({FIELD, PARAMETER})
+@Target({ METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER, TYPE_USE })
```
Comment (JoergSiebahn) [inline sda-commons-server-key-mgmt/src/main/java/org/sdase/commons/keymgmt/validator/PlatformKey.java:26]: Is there anything that can be validated at the constructor or are the Hibernate docs missing something?

```suggestion
@Target({ FIELD, METHOD, PARAMETER, ANNOTATION_TYPE, TYPE_USE })
```
Taken from here: https://docs.jboss.org/hibernate/stable/validator/reference/en-US/html_single/?v=7.0#validator-customconstraints-constraintannotation

Note: Spotless strikes back again.

Comment (ahoeing) [reply sda-commons-server-key-mgmt/src/main/java/org/sdase/commons/keymgmt/validator/PlatformKey.java:26]: I took the list from org.hibernate.validator.internal.constraintvalidators.bv.number.sign.NegativeOrZeroValidatorForBigInteger but I am fine with the suggestion

**Review by christopher-cudennec — COMMENTED** [review]
Do you want to add some tests to make sure all targets are supported and work as expected?

**Comment by ahoeing** [comment]
I added the tests for the options used by dropwizard. I think it is sufficient.

**Review by christopher-cudennec — APPROVED** [review]

**Pull request merged**